### ISSUE TITLE
New variables to be tested

### ISSUE THREAD

**Issue by mverzett** [cms-btv-pog/CTagTraining#1]
Hi @gvonsem @smoortga I thought we can keep track here of the variables we want to include in the next ntuple production. Here is a list of what I remember from our discussion. Feel free to extend it.
- corrected vertex mass as in [here](https://indico.cern.ch/event/439699/contribution/3/attachments/1143670/1638978/cjets_HeavyIons_kjung_150824.pdf)
- [quark-gluon](https://cds.cern.ch/record/1599732/files/JME-13-002-pas.pdf) discriminating variables
- relative energy (pt) deposits in concentric rings around the jet axis (0.5 or 1 DR)

**Comment by smoortga** [comment]
These variables indeed seem promising (or at least worthwhile to check)!
I can't think of any others for the moment.

Cheers,

Seth

**Comment by pvmulder** [comment]
I would also suggest to read in detail this note:
https://atlas.web.cern.ch/Atlas/GROUPS/PHYSICS/PUBNOTES/ATL-PHYS-PUB-2015-001/
maybe we can get some inspiration from it. 
Please note that Sarmad and Francesco are also developing new (b-tag) variables. Some of these may be interesting for c-tagging as well. They will present an update today.

**Comment by smoortga** [comment]
Thanks Petra,

I already had a quick look at this paper during my thesis.

Perhaps one interesting feature that immediately caught my attention was the fact that ATLAS combines all the Impact Parameter-related values in a Likelihood ratio (see Table 1 row "IP3D" in the paper Petra suggested) and uses this as 1 parameter. I have no clue whether or not this will really improve the performance, but it would reduce the amount of parameters in any case and the likelihood discriminator might be more discriminating and easier to handle by the BDT than all the IP variables seperately...

Another feature that I noticed is that ATLAS uses the Minimum, Maximum and Mean track rapidity, whereas we take the rapidity (or pseudo-rapidity) of the three most displaced tracks.

And finally they have some variables (n_vs, n_trk^JF, n_1t) related to SV's with a few tracks only (see same table in the row JetFitter), which I don't think we use. But for this we should check which vertices we reconstruct because there might be a selection requiring at least 2 tracks or something... I don't remember.

So to summarize, ATLAS uses these features which we don't use I think:

1) combine IP in likelihood ratio
2) Max, Min and Mean eta instead of 3 most displaced tracks
3) Some SV related related variables with a few tracks only

**Comment by pvmulder** [comment]
Concerning 1) combine IP in likelihood ratio:
From what I understand, they do not use IP3D directly, but a 2D distribution of IP2Dsig (transverse IP significance) and IP1Dsig (longitudinal IP significance). While IP1Dsig is more sensitive to PU, they exploit optimally the correlation between the transverse and longitudinal IP significances. Combining the information from the IP significances of all tracks is indeed a good idea and hard to do in a BDT because the number of tracks associated to the jet may be different in each jet and default values need to be assigned. However, in some sense this is what is done also in our JetProbability tagger (not exploiting the tranverse versus longitudinal aspect however). It would be interesting to investigate how much the longitudinal IP significance could add (something that Sarmad and Francesco investigate)

2) Max, Min and Mean eta instead of 3 most displaced tracks
This is indeed something to explore, but there are more differences. They are also combining the information of all vertices (and they use the total vertex flight-length significance) instead of using the information of the first vertex in the list (what we do), they use for instance also the transverse displacement of the tertiary vertex

3) Some SV related related variables with a few tracks only:
In this context it should be noted that a SV that is fitted requires 2 tracks. However, in the JetFitter approach, they define single-track vertices based on a track and the fitted flight line of the B and/or D meson. We had a similar approach in the past with the GhostTrack tagger and we should re-investigate this approach and corresponding variables. 

Clearly there is much to gain and many options to investigate. Because in addition to the above, I am confident that we can also optimize the track selection itself for D mesons. We will have a presentation today as well on this topic (in the context of b-tagging) by Brieuc and Sebastien
